src/jestMatchers.ts

```diff
@@ -0,0 +1,338 @@
+        input: Record<string, unknown>
+    ) {
+        return processMatch({
+            ctx: this,
+            mockClient,
+            command,
+            input,
+            check: ({ commandCalls }) => ({ pass: commandCalls.length > 0, data: {} }),
+            message: ({ client, cmd, calls }) => [
+                `Expected ${client} have received ${cmd} with ${this.utils.printExpected(
+                    input
+                )}`,
+                `But ${this.utils.printReceived(0)} have been received`,
+                'Calls:',
+                '',
+                ...printCalls(this, calls),
+            ],
+            notMessage: ({ client, cmd, commandCalls }) => [
+                `Expected ${client} Not have received ${cmd} with ${this.utils.printExpected(
+                    input
+                )}`,
+                `But ${this.utils.printReceived(
+                    commandCalls.length
+                )} have been received`,
+                'Calls:',
+                '',
+                ...printCalls(this, commandCalls),
+            ],
+        });
+    },
+    /**
+     * implementation of {@link AwsSdkJestMockMatchers.toHaveNthReceivedCommandWith} matcher
+     */
+    toHaveNthReceivedCommandWith(
+        this: jest.MatcherContext,
+        mockClient: ClientMock,
+        call: number,
+        command: new () => AnyCommand,
+        input?: Record<string, unknown>
+    ) {
+        assert(
+            call && typeof call === 'number' && call > 0,
+            'Call number must be a number and greater as 0'
+        );
+
+        return processMatch<{ received: AnyCommand; cmd: string }>({
+            ctx: this,
+            mockClient,
+            command,
+            check: ({ calls }) => {
+                const received = calls[call - 1].args[0];
+                return {
+                    pass:
+                        received instanceof command && this.equals(received.input, input),
+                    data: {
+                        received,
+                        cmd: this.utils.printReceived(received.constructor.name),
+                    },
+                };
+            },
+            input,
+            message: ({ cmd, client, calls, data }) => [
+                `Expected ${client} have ${call} received ${cmd}`,
+                `Received ${data.cmd} with input`,
+                this.utils.printDiffOrStringify(
+                    input,
+                    data.received.input,
+                    'Expected',
+                    'Received',
+                    false
+                ),
+                'Command Calls:',
+                '',
+                ...printCalls(this, calls),
+            ],
+            notMessage: ({ cmd, client, data }) => [
+                `Expected ${client} Not have ${call} received ${cmd} with matching input`,
+                `But ${data.cmd} have been received matching given input`,
+            ],
+        });
+    },
```

Comment:
Please make similar changes to all the messages as suggested in `toHaveReceivedCommandTimes`.